# Question 1 - How Many Unique Campaigns Ran in February?


In [21]:
# IMPORT LIBRARIES AND MODULES

import numpy as np
import pandas as pd 
from pandas import Series

In [22]:
# READ IN FILES

source_1 = pd.read_csv('source1.csv')
source_2 = pd.read_csv('source2.csv')

In [23]:
# CREATE DATAFRAME WITH ONLY CAMPAIGN AND DATE VALUES

campaign_and_date = source_1[['campaign', 'date']]
print(campaign_and_date.head(10))

          campaign     date
0  fish_cow_desert   1/1/15
1  fish_cow_desert   1/2/15
2  fish_cow_desert   1/3/15
3  fish_cow_desert   1/4/15
4  fish_cow_desert   1/5/15
5  fish_cow_desert   1/6/15
6  fish_cow_desert   1/7/15
7  fish_cow_desert   1/8/15
8  fish_cow_desert   1/9/15
9  fish_cow_desert  1/10/15


In [43]:
# FILTER OUT NON-FEBRUARY DATES IN THE 'date' SERIES

campaigns_in_feb = campaign_and_date[campaign_and_date['date'].str.contains('^2')]
print(campaigns_in_feb.head(10))
print(len(campaigns_in_feb))

           campaign     date
31  fish_cow_desert   2/1/15
32  fish_cow_desert   2/2/15
33  fish_cow_desert   2/3/15
34  fish_cow_desert   2/4/15
35  fish_cow_desert   2/5/15
36  fish_cow_desert   2/6/15
37  fish_cow_desert   2/7/15
38  fish_cow_desert   2/8/15
39  fish_cow_desert   2/9/15
40  fish_cow_desert  2/10/15
3500


In [48]:
# STORE AND PRINT UNIQUE CAMPAIGNS IN FEBRUARY

unique_campaigns_in_feb = campaigns_in_feb['campaign'].unique()
print(unique_campaigns_in_feb[:20])
print(len(unique_campaigns_in_feb))

['fish_cow_desert' 'fish_cow_jungle' 'fish_cow_mountains' 'fish_cow_plains'
 'fish_cow_valley' 'fish_dog_desert' 'fish_dog_jungle' 'fish_dog_mountains'
 'fish_dog_plains' 'fish_dog_valley' 'fish_lion_desert' 'fish_lion_jungle'
 'fish_lion_mountains' 'fish_lion_plains' 'fish_lion_valley'
 'fish_monkey_desert' 'fish_monkey_jungle' 'fish_monkey_mountains'
 'fish_monkey_plains' 'fish_monkey_valley']
125


### This concludes that there were <span style="color:green">125</span> unique campaigns in the month of February.

# Question 2 - What is the Total Number of Conversions on Plants?

In [60]:
# CONDENSE THE DATAFRAME DOWN TO CAMPAIGNS CONTAINING PLANTS

plants_campaigns = source_1[source_1['campaign'].str.contains('plants')]
print(plants_campaigns.head(5))

               campaign    date  spend  impressions  \
7500  plants_cow_desert  1/1/15  57.61         6581   
7501  plants_cow_desert  1/2/15  37.55         5031   
7502  plants_cow_desert  1/3/15  48.83         9731   
7503  plants_cow_desert  1/4/15  15.80         1792   
7504  plants_cow_desert  1/5/15  57.20         9060   

                                                actions  
7500  [{"action": "conversions", "c": 47}, {"action"...  
7501  [{"action": "clicks", "z": 47}, {"action": "cl...  
7502                    [{"a": 60, "action": "clicks"}]  
7503  [{"y": 47, "action": "conversions"}, {"action"...  
7504  [{"y": 12, "action": "conversions"}, {"action"...  


In [89]:
print(type(plants_campaigns['actions']))


<class 'pandas.core.series.Series'>
